squirrel.frontier/src/test/java/org/dice_research/squirrel/queue/domainbased/MongoDBDomainQueueTest.java

```diff
@@ -162,4 +232,84 @@ public void deleteUris() throws Exception {
+        uriList.add(uri1);
+        uriList.add(uri2);
+        uriList.add(uri3);
+        uriList.add(uri4);
+        uriList.add(uri5);
+        uriList.add(uri6);
+        Map<String, List<CrawleableUri>> keywiseUris = mongodbQueue.makeURIsKeywise(uriList);
+        Assert.assertEquals(3, keywiseUris.get("dbpedia.org").size());
+        Assert.assertEquals(2, keywiseUris.get("en.wikipedia.org").size());
+        Assert.assertEquals(1, keywiseUris.get("www.lonelyplanet.com").size());
+        Assert.assertTrue(keywiseUris.get("dbpedia.org").contains(uri1));
+        Assert.assertTrue(keywiseUris.get("dbpedia.org").contains(uri2));
+        Assert.assertTrue(keywiseUris.get("dbpedia.org").contains(uri3));
+        Assert.assertTrue(keywiseUris.get("en.wikipedia.org").contains(uri4));
+        Assert.assertTrue(keywiseUris.get("en.wikipedia.org").contains(uri5));
+        Assert.assertTrue(keywiseUris.get("www.lonelyplanet.com").contains(uri6));
+    }
+
+    @Test
+    public void testAddKeywiseUrisForFirstTime() throws URISyntaxException {
+        mongodbQueue.open();
+        mongodbQueue.purge();
+        Map<String, List<CrawleableUri>> keyWiseUris = new HashMap<>();
+        List<CrawleableUri> dbpediaUris = new ArrayList<>();
+        CrawleableUri uri1 = new CrawleableUri(new URI("http://dbpedia.org/resource/Paderborn"));
+        CrawleableUri uri2 = new CrawleableUri(new URI("http://dbpedia.org/resource/Sirsi"));
+        CrawleableUri uri3 = new CrawleableUri(new URI("http://dbpedia.org/resource/Cologne"));
+        dbpediaUris.add(uri1);
+        dbpediaUris.add(uri2);
+        dbpediaUris.add(uri3);
+        keyWiseUris.put("dbpedia.org", dbpediaUris);
+        List<CrawleableUri> urisNotAdded = mongodbQueue.addKeywiseUris(keyWiseUris, 2);
+        Assert.assertEquals(0, urisNotAdded.size());
+        mongodbQueue.purge();
+        mongodbQueue.close();
+    }
+
+    @Test
+    public void testAddKeywiseUrisWithBelowCriticalScores() throws URISyntaxException {
+        mongodbQueue.open();
+        mongodbQueue.purge();
+        Map<String, List<CrawleableUri>> keyWiseUris = new HashMap<>();
+        List<CrawleableUri> dbpediaUris = new ArrayList<>();
+        CrawleableUri uri1 = new CrawleableUri(new URI("http://dbpedia.org/resource/Berlin"));
+        CrawleableUri uri2 = new CrawleableUri(new URI("http://dbpedia.org/resource/Bangalore"));
+        CrawleableUri uri3 = new CrawleableUri(new URI("http://dbpedia.org/resource/New_York_City"));
+        mongodbQueue.addUri(uri1);
+        mongodbQueue.addUri(uri2);
+        mongodbQueue.addUri(uri3);
+        dbpediaUris.add(uri1);
+        dbpediaUris.add(uri2);
+        dbpediaUris.add(uri3);
+        keyWiseUris.put("dbpedia.org", dbpediaUris);
+        List<CrawleableUri> urisNotAdded = mongodbQueue.addKeywiseUris(keyWiseUris, 2);
+        Assert.assertEquals(3, urisNotAdded.size());
+        mongodbQueue.purge();
+        mongodbQueue.close();
```

Comment:
This test should ask the queue for URIs. If the test case would have this, I am sure that it would reveal that the URIs still have been added because of the bug in the workflow described in one of my previous comments.